lib/service_actor/core.rb

```diff
@@ -54,4 +54,13 @@ def _call
   def fail!(**arguments)
     result.fail!(**arguments)
   end
+
+  private
+
+  # Raises an error depending on the mode
+  def raise_error_with(message, **arguments)
+    raise ServiceActor::ArgumentError, message unless message.is_a?(Proc)
+
+    raise ServiceActor::ArgumentError, message.call(*arguments.values)
```

Comment:
Minor change here but we can reduce duplication slightly:

```suggestion
    message = message.call(*arguments.values) unless message.is_a?(Proc)

    raise ServiceActor::ArgumentError, message
```